### ISSUE TITLE
rpm --restore/--setperms/ --setugids/--setcaps should ignore ghost files by default

### ISSUE THREAD

**Issue by maage** [rpm-software-management/rpm#591]
Currently as far as I can see in many cases ghost files are not marked fully in spec files. Problem is when ghost file does not exist but is defined at spec file.

If you run rpm --restore you can change many files to perms like d---------. or ----------.

Another way to fix this is to require ghost files to have full set of metadata when they are defined at spec or disable feature where ghost files can be defined without actual entity.

I'd conside --restore to be dangerous option currently.

[Removed wrong commend about creation]

**Comment by pmatilai** [comment]
Hmm, explicit attr used to be required for ghosts that don't exist in the buildroot, but seems that went out with d6775a746bbaf8ddc8a91c9fae68ae91af1cc570. In retrospective, "If %defattr() doesn't specify a mode the file will be seen without any permissions at all, but that's not strictly an error (and same can happen with %dev() already)" seems a bit questionable.

**Comment by opoplawski** [comment]
Also, the files may not exist:
```
$ rpm --setperms ipa-server-common-4.6.5-11.sl7_7.4.noarch
chmod: cannot access ‘/etc/httpd/conf.d/ipa-pki-proxy.conf’: No such file or directory
chmod: cannot access ‘/etc/systemd/system/httpd.d/ipa.conf’: No such file or directory
chmod: cannot access ‘/var/lib/ipa/pki-ca/publish’: No such file or directory
chmod: cannot access ‘/var/lib/kdcproxy’: No such file or directory
chmod: cannot access ‘/var/named/dyndb-ldap/ipa’: No such file or directory
$ echo $?
1
```

**Comment by pmatilai** [comment]
Should be dealt with as a part of #965

**Comment by pmatilai** [comment]
...and really fixed in commit 2a8d89a1c390e6ca1de0ece49aa11012b1d0ff89.
